### DISCUSSION TITLE
Announcing: EDGAR

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#345] [General]
1) Apologies for any unnecessary noise this may raise.

2) I wanted to share my first program written (almost) from scratch: EDGAR - Link: https://github.com/aykaramba/edgar

3) EDGAR is a single page application programmed in CLOG + Common Lisp.  This was a self learning exercise to see if I can build a basic UI for a set of tutorials that I was working through here: https://medium.com/@codingcamel/extracting-fundamental-stock-data-from-edgar-using-our-favorite-language-common-lisp-part-1-a3171d100dd4

The goal was to create a simple UI that allows a user to download the corporate filings from the SEC EDGAR database and display a paginated view of the 10 000+ listed companies, a view to display the data downloads, a popup to view the values of various variables for troubleshooting and view to display some help information.

A nice simple, easy to build and easy to package app.

4) What I have learned:

a) Prefix notation is the syntax that just clicks with me.  Infix syntax is truly what has kept me from enjoying writing a single line of code.  The combination of editors like LEM or Emacs and the Paredit mode are a from another world.  Being able to basically drag and  drop forms (regardless of their complexity) around a program is amazing.  With access to the REPL and a simple C-c on every form they work on, a student can iterate through the learning process quickly and in a meaningful way that helps the student retain what they have learned.

b) CLOG - CLOG is easy to use because it is discoverable.  Once you see the pattern you can usually guess how any particular component works and it has been a joy to use.  In addition, pairing CLOG with W3.CSS instead of the default of something like Bootstrap is an an extremely tasteful choice.  W3.CSS is pretty simple but probably as powerful as it needs to be for most project that I have been involved with over the years.  The combination of CLOG + W3.CSS cannot be undersold in my opinion.  I may be a beginner programmer but right now I can sit down and mock up an entire interface and basic functionality in CLOG +W3.CSS without much thinking, it is that straight forward for basic CRUD applications.  

c) CLOG tutorials, clog manuals, clog example files + projects and Rabbi Botton patiently answering questions has also been an incredibly useful set of tools to help onboard a noob like me into the the programming world.   David, I greatly appreciate you creating CLOG and providing your valuable time in so many dimensions to help out. 

d) I am grateful for everyone on the IRC channel for also patiently answering my questions.  They also have helped lower the barrier to entry for me.

5) SUMMARY: I think I have finally caught the programming bug.   Everything is a joy to use and CLOG lowers the barrier to entry for students to learn programming in Common Lisp and it makes it great fun to whiteboard an idea directly in code and see where that leads.  

Any and all critiques are welcome, particularly around code style and indentation.  I looked up the accepted norms, not sure if I got everything right.

Thanks everyone!

**Comment by tluyben** [comment]
Well done! I would suggest adding some screenshots in the GitHub repository to get a quick idea!

**Reply by aykaramba** [reply]
Aha!  Communication, presentation, packaging and all that.  Got it.  Done. 

Thanks!

**Comment by rabbibotton** [comment]
Fantastic :) Maybe share on https://www.reddit.com/r/lisp/ too :)
